### PR TITLE
Bump to v0.4.0

### PR DESCRIPTION
**Pull request by overvenus**
Waits for #166 and #224 .

### CODE REVIEW

**Comment by siddontang** [comment]
I think we don't need to wait these PRs, upgrading to 1.14 is a huge change, we can release a version directly.

**Comment by overvenus** [comment]
PTAL @siddontang and @BusyJay

File: proto/Cargo.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ build = "build.rs"
 [dependencies]
 protobuf = "~2.0"
 futures = "0.1"
-grpcio = { path = "..", version = "0.3.0" }
+grpcio = { path = "..", version = "0.4.0" }
 
 [build-dependencies]
```
Comment (messense) [inline proto/Cargo.toml]: `grpcio-compiler = { path = "../compiler", version = "0.3.0" }` below should be changed to `grpcio-compiler = { path = "../compiler", version = "0.4.0" }` I think.

Comment (overvenus) [reply proto/Cargo.toml]: You are right!

**Comment by siddontang** [comment]
CI failed

File: README.md [hunk 2]
```diff
@@ -11,13 +11,13 @@
...
 
-- [x] Basic asynchronous unary/steaming call 
+- [x] Basic asynchronous unary/steaming call
 - [x] SSL
 - [x] Generic call
 - [x] Connection level compression
 - [x] Interoperability test
 - [x] QPS benchmark
-- [ ] Custom metadata
+- [x] Custom metadata
```
Comment (BusyJay) [inline README.md]: This is not fully supported.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+# 0.4.0 - 2018-09-15
+
+- Update gRPC from 1.7.2 to 1.14.2
+- Services accept mut reference
+- Cancel RPC when senders and receivers were dropped
```
Comment (BusyJay) [inline CHANGELOG.md]: Should mention the changes to executor.

**Pull request merged**